packages/backend/server/src/base/mailer/mail.service.ts

```diff
@@ -45,318 +65,214 @@ export class MailService {
     return !!this.mailer;
   }
 
-  async sendInviteEmail(
-    to: string,
-    inviteId: string,
-    invitationInfo: {
-      workspace: {
-        id: string;
-        name: string;
-        avatar: string;
-      };
-      user: {
-        avatar: string;
-        name: string;
-      };
-    }
-  ) {
-    const buttonUrl = this.url.link(`/invite/${inviteId}`);
-    const workspaceAvatar = invitationInfo.workspace.avatar;
-
-    const content = `<p style="margin:0">${
-      invitationInfo.user.avatar
-        ? `<img
-    src="${invitationInfo.user.avatar}"
-    alt=""
-    width="24px"
-    height="24px"
-    style="width:24px; height:24px; border-radius: 12px;object-fit: cover;vertical-align: middle"
-  />`
-        : ''
-    }
-  <span style="font-weight:500;margin-right: 4px;">${
-    invitationInfo.user.name
-  }</span>
-  <span>invited you to join</span>
-  <img
-    src="cid:workspaceAvatar"
-    alt=""
-    width="24px"
-    height="24px"
-    style="width:24px; height:24px; margin-left:4px;border-radius: 12px;object-fit: cover;vertical-align: middle"
-  />
-  <span style="font-weight:500;margin-right: 4px;">${
-    invitationInfo.workspace.name
-  }</span></p><p style="margin-top:8px;margin-bottom:0;">Click button to join this workspace</p>`;
-
-    const html = emailTemplate({
-      title: 'You are invited!',
-      content,
-      buttonContent: 'Accept & Join',
-      buttonUrl,
-    });
+  async sendSignUpMail(to: string, url: string) {
```

Comment:
we can integrate mailer with event system, so we can decouple it from business services

Reply:
the sending results of some mails need to be returne to the frontend for success/failure notifications
so not all deps from bussiness modules can be remove utils the notification system is completed